File: data/EXP01/python/2406/2406.output_1.py
# @lc app=leetcode id=2406 lang=python3
from typing import List

class Solution:
    def minGroups(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda x: x[0])  # Sort intervals based on starting point
        groups = 1  # Initialize the number of groups

        end = intervals[0][1]  # Initialize the end point of the first interval
        for i in range(1, len(intervals)):
            if intervals[i][0] > end:  # If the interval does not overlap with the previous one
                groups += 1  # Start a new group
                end = intervals[i][1]  # Update the end point
            else:
                end = max(end, intervals[i][1])  # Update the end point if the interval overlaps
        
        return groups

File: data/EXP01/golang_no_system_var/2406/2406.output_1.go
func minGroups(intervals [][]int) int {
    if len(intervals) == 0 {
        return 0
    }
  
    sort.Slice(intervals, func(i, j int) bool {
        return intervals[i][0] < intervals[j][0]
    })
  
    groups := 1
    maxEnd := intervals[0][1]
  
    for i := 1; i < len(intervals); i++ {
        if intervals[i][0] > maxEnd {
            groups++
            maxEnd = intervals[i][1]
        } else {
            maxEnd = max(maxEnd, intervals[i][1])
        }
    }
  
    return groups
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}
